Automattic/jetpack
# Contact Info Block add manual blur to fix focus issues

Fixes bug where selecting and typing in an inner block on Contact Info and then exiting out of the block fast (clicking on a different unnested block) would cause cursor to stay in the original inner block and thus result in an unhideable keyboard.

Fixes [#3122](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3122)

Gutenberg-Mobile: [#3340](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3340)

NOTE: Codeclimate mentions that the component (function) is too long/complex; however, since I refactored the class component into a functional one, the entire component itself is a function. I tried refactoring out subcomponents but found that the code was harder to read. I'm going to ignore these suggestions as not relevant to this code.

#### Changes proposed in this Pull Request:

* Adds manual `blur()` call on text input if block is no longer selected.

#### Does this pull request change what data or activity we track or use?
No change to data/tracked activity.

#### Add changelog entries to affected projects
Added under `changelog/fix-contact-info-focus`.

#### Screenshots
<details>

| Before | After |
| - | - |
| <img src="https://user-images.githubusercontent.com/13263478/113199110-5bb90300-922c-11eb-8f19-2959aaf00979.gif" width="100%"/> | <img src="https://user-images.githubusercontent.com/13263478/113197601-7ab69580-922a-11eb-806a-130902c89f09.gif" width="50%"/> |
</details>


#### Testing instructions:
* Have Jetpack-enabled site with version > 8.5.
* Create new post
* Add contact info block
* Type 'test' into "Address Line 2"
* Quickly click on "Email"
* See that cursor is removed from "Address Line 2" and keyboard hides.
